### ISSUE TITLE
Will already downsampled blocks be merged?When downsampled blocks are merged into new blocks, do the new blocks still need to be downsampled?

### ISSUE THREAD

**Issue by lu-xiansheng** [thanos-io/thanos#7554]
**Thanos, Prometheus and Golang version used**:

**Object Storage Provider**:

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Full logs to relevant components**:

**Anything else we need to know**:

**Comment by yeya24** [comment]
Yeah I think the current logic is bugged in previous version of Thanos. But with https://github.com/thanos-io/thanos/pull/7334, downsampled blocks will be excluded from compaction so they shouldn't be compacted anymore.
